src/acmp/mod.rs

```diff
@@ -0,0 +1,157 @@
+pub struct Init {
+    pub full_bias:                   bool,
+    pub half_bias:                   bool,
+    pub bias_prog:                   u32,
+    pub interrupt_on_falling_edge:   bool,
+    pub interrupt_on_rising_edge:    bool,
+    pub warm_time:                   WarmTime,
+    pub hysteresis_level:            HysteresisLevel,
+    pub inactive_value:              bool,
+    pub low_power_reference_enabled: bool,
+    pub vdd_level:                   u32,
+    pub enable:                      bool,
+}
+
+impl Default for Init {
+    fn default() -> Init {
+        Init {
+            full_bias:                   false,
+            half_bias:                   false,
+            bias_prog:                   0x7,
+            interrupt_on_falling_edge:   false,
+            interrupt_on_rising_edge:    false,
+            warm_time:                   WarmTime::_512,
+            hysteresis_level:            HysteresisLevel::_5,
+            inactive_value:              false,
+            low_power_reference_enabled: false,
+            vdd_level:                   0x3D,
+            enable:                      true,
+        }
+    }
+}
+
+#[allow(non_snake_case)]
+#[repr(C)]
+#[derive(Copy)]
+pub struct Acmp {
+    pub CTRL:     u32,
+    pub INPUTSEL: u32,
+    pub STATUS:   u32,
+    pub IEN:      u32,
+    pub IF:       u32,
+    pub IFS:      u32,
+    pub IFC:      u32,
+    pub ROUTE:    u32,
+}
+
+impl Acmp {
+    #[inline]
+    pub fn acmp0() -> &'static mut Acmp {
+        unsafe { transmute(GET_ACMP0()) }
+    }
+
+    #[inline]
+    pub fn acmp1() -> &'static mut Acmp {
+        unsafe { transmute(GET_ACMP1()) }
+    }
+
+    pub fn init(&self, init: &Init) {
+        unsafe { ACMP_Init(self, init) }
+    }
+
+    pub fn gpio_setup(&self, location: u32, enable: bool, invert: bool) {
+        unsafe { ACMP_GPIOSetup(self, location, enable, invert) }
+    }
+
+    pub fn channel_set(&self, neg_sel: Channel, pos_sel: Channel) {
+        unsafe { ACMP_ChannelSet(self, neg_sel, pos_sel) }
+    }
+}
+
+#[repr(u8)]
+#[derive(Copy)]
+pub enum WarmTime {
+    _4   = 0x0,
+    _8   = 0x1,
+    _16  = 0x2,
+    _32  = 0x3,
+    _64  = 0x4,
+    _128 = 0x5,
+    _256 = 0x6,
+    _512 = 0x7,
+}
+
+#[repr(u8)]
+#[derive(Copy)]
+pub enum HysteresisLevel {
+    _0 = 0x0,
+    _1 = 0x1,
+    _2 = 0x2,
+    _3 = 0x3,
+    _4 = 0x4,
+    _5 = 0x5,
+    _6 = 0x6,
+    _7 = 0x7,
+}
+
+#[repr(C)]
+#[derive(Copy)]
+pub struct CapsenseInit {
+    pub full_bias:                   bool,
+    pub half_bias:                   bool,
+    pub bias_prog:                   u32,
+    pub warm_time:                   WarmTime,
+    pub hysteresis_level:            HysteresisLevel,
+    pub resistor:                    CapsenseResistor,
+    pub low_power_reference_enabled: bool,
+    pub vdd_level:                   u32,
+    pub enable:                      bool,
+}
+
+#[repr(u8)]
+#[derive(Copy)]
+pub enum CapsenseResistor {
+    Resistor0 = 0x0,
```

Comment:
Why not `_0` ? Well we need to decide a common way of naming these and stick to it. Have you seen some examples in the rusty wild?